File: golang/rabbitsinforest781/rabbits-in-forest-781.go
package main

import (
	"fmt"
	"math"
)

func main() {
	answers := []int{10, 10, 10}
	fmt.Println(numRabbits(answers))
}

func numRabbits(answers []int) int {
	m := map[int]int{}
	for _, v := range answers {
		if _, ok := m[v]; ok {
			m[v] = m[v] + 1
		} else {
			m[v] = 1
		}
	}

	count := 0

	for k, v := range m {
		if k+1 == v {
			count += k + 1
		} else if k+1 > v {
			count += k + 1
		} else {
			count += (k + 1) * int(math.Ceil(float64(v)/float64(k+1)))
		}
	}

	return count
}

File: nodejs/rabbits-in-forest-781.js
'use strict';
/**
 * @param {number[]} answers
 * @return {number}
 */
var numRabbits = function(answers) {
    let map = new Map();

    for (let v of answers) {
        if (map.has(v)) map.set(v,map.get(v)+1);
        else map.set(v,1);
    }
    let count = 0;
    for (var [k,v] of map) {
        if (k+1>=v) {
            count += k+1
        } else {
            count += (k+1)*Math.ceil(v/(k+1));
        }
    }
    return count;
};

console.log(numRabbits([3, 3, 3, 3,3]));